cf/commands/domain/create_shared_domain_test.go

```diff
@@ -1,73 +1,308 @@
+				})
+			})
+
+			Context("when router-group flag is not set", func() {
+				BeforeEach(func() {
+					flagContext.Parse("domain-name")
+				})
+
+				It("returns a LoginRequirement", func() {
+					actualRequirements, err := cmd.Requirements(factory, flagContext)
+					Expect(err).NotTo(HaveOccurred())
+					Expect(factory.NewLoginRequirementCallCount()).To(Equal(1))
+					Expect(actualRequirements).To(ContainElement(loginRequirement))
+				})
+
+				It("does not return a RoutingApiRequirement", func() {
+					actualRequirements, err := cmd.Requirements(factory, flagContext)
+					Expect(err).NotTo(HaveOccurred())
+					Expect(factory.NewRoutingAPIRequirementCallCount()).To(Equal(0))
+					Expect(actualRequirements).ToNot(ContainElement(routingApiRequirement))
+				})
+
+				It("does not return a MinAPIVersionRequirement", func() {
+					actualRequirements, err := cmd.Requirements(factory, flagContext)
+					Expect(err).NotTo(HaveOccurred())
+					Expect(actualRequirements).NotTo(ContainElement(minAPIVersionRequirement))
+				})
+			})
+		})
 	})
 
-	It("TestShareDomain", func() {
-		runCommand("example.com")
+	Describe("Execute", func() {
+		Context("when router-group flag is not set", func() {
+			BeforeEach(func() {
+				flagContext.Parse("domain-name")
+				cmd.Execute(flagContext)
+			})
+
+			It("prints a message", func() {
+				Expect(ui.Outputs).To(ContainSubstrings(
+					[]string{"Creating shared domain domain-name"},
+				))
+			})
+
+			It("creates a shared domain", func() {
+				Expect(domainRepo.CreateSharedDomainCallCount()).To(Equal(1))
+
+				domainName, routerGroupGuid := domainRepo.CreateSharedDomainArgsForCall(0)
+				Expect(domainName).To(Equal("domain-name"))
+				Expect(routerGroupGuid).To(Equal(""))
+			})
+
+			It("prints success message", func() {
+				Expect(ui.Outputs).To(ContainSubstrings(
+					[]string{"OK"},
+				))
+			})
+
+		})
+
+		Context("when cannot create shared domain", func() {
+			BeforeEach(func() {
+				flagContext.Parse("domain-name")
+				domainRepo.CreateSharedDomainReturns(errors.New("create-domain-error"))
+			})
+
+			It("fails with error", func() {
+				Expect(func() { cmd.Execute(flagContext) }).To(Panic())
+				Expect(ui.Outputs).To(ContainSubstrings(
+					[]string{"FAILED"},
+					[]string{"create-domain-error"},
+				))
+			})
+		})
+
+		Context("when router-group flag is set", func() {
+			BeforeEach(func() {
+				flagContext.Parse("domain-name", "--router-group", "router-group-name")
+			})
+
+			It("retrieves a list of router groups from the Routing Api", func() {
+				cmd.Execute(flagContext)
+				Expect(routingApiRepo.ListRouterGroupsCallCount()).To(Equal(1))
+			})
+
+			Context("when routing group is found", func() {
+				BeforeEach(func() {
+					cmd.Execute(flagContext)
+				})
+
+				It("prints a message", func() {
+					Expect(ui.Outputs).To(ContainSubstrings(
+						[]string{"Creating shared domain domain-name"},
+					))
+				})
+
+				It("creates a shared domain", func() {
```

Comment:
This should have the form of `it tries to`. There are also tests missing for when the attempt to create the shared domain fails.